spec/factories/ect_at_school_period_factory.rb

```diff
@@ -25,6 +25,7 @@
 
     trait :school_led do
       programme_type { 'school_led' }
+      lead_provider { nil }
```

Comment:
@ltello do you think it is a good idea to set the `lead_provider` to `nil` in this factory, since we are now validating that it's `nil`:

https://github.com/DFE-Digital/register-early-career-teachers-public/blob/b9ebeb4b0394c4778e45a663b5f561b1721ebb14/app/models/ect_at_school_period.rb#L50-L54

If this is not the approach we want to take globally, I can set it to `nil` in my specific test. 

Keen to hear your thoughts!

Reply:
I think generally you want factories to construct valid objects so in this instance I think it's fine, a school-led training  period won't have a lead provider.